src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -35,6 +44,15 @@ def create
 
   private
 
+  def set_token
+    @token = ::TriggerControllerService::TokenExtractor.new(request).call
+    raise InvalidToken, 'No valid token found' unless @token
+  end
+
+  def pundit_user
+    @token.executor
+  end
```

Comment:
We can extract those two methods into a module and share it with `TriggerController` also

Reply:
To achieve DRYness. The next time someone needs to change those methods they will start to diverge. Having them in a single place will make people think about all the use cases.

Reply:
If that "next time" comes we can optimize it. Or in other words: f**k the future :)

Reply:
All right, then, f**ck the future.